File: maximum subarray/maxSub.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]
        currentSum = 0
        for n in nums:
            if currentSum < 0 :
                currentSum = 0
            currentSum += n
            maxSub = max(maxSub,currentSum)
        return maxSub

File: maximum subarray/maxSub.js
const solution = (array) => {
    let currentSum = 0
    let maxSub = array[0]
    array.forEach((item) => {
        if (currentSum < 0) {
            currentSum = 0
        }
        currentSum += item
        maxSub = Math.max(maxSub, currentSum)
    })
    return maxSub
}


console.log(solution([-2, 1, -3, 4, -1, 2, 1, -5, 4]))